src/utils/hooks/useUndoQueue.tsx

```diff
@@ -0,0 +1,55 @@
+import { useState } from 'react'
+
+export const useUndoQueue = <T,>(initialValue?: T) => {
```

Comment:
Don't bother reviewing this, as I've replaced it with a 3rd-party package in #1514